dask/dataframe/io/hdf.py

```diff
@@ -309,8 +315,9 @@ def read_hdf(pattern, key, start=0, stop=None, columns=None,
         raise ValueError("When assuming pre-partitioned data, data must be "
                          "read in its entirety using the same chunksizes")
     from ..multi import concat
-    return concat([_read_single_hdf(path, key, start=start, stop=stop,
-                                    columns=columns, chunksize=chunksize,
-                                    sorted_index=sorted_index,
-                                    lock=lock, mode=mode)
-                   for path in paths])
+    return concat(list(compute(*[
+        delayed(_read_single_hdf)(path, key, start=start, stop=stop,
+                                  columns=columns, chunksize=chunksize,
+                                  sorted_index=sorted_index,
+                                  lock=lock, mode=mode)
```

Comment:
This confuses me.  `_read_single_hdf` already produces a dask.dataframe.  Why are we delaying this function?